### PR TITLE
MOD: Changing monstergroup to remove very advanced zombie monsters from the beginings of the game

### LINKED ISSUE #23063: Wander spawn monsters don't follow evolution rules
Game version:0.C-26482-g5d60a3e

Operating system:Windows 7

Tiles or curses:curses

Mods active:Filthy clothing, disable NPC needs, simplified nutrition, generic guns, more survival tools, makeshift items, mining, cataclysm++, stats through skills IV, extra professions, classes and scenarios

### Expected behavior
Wander spawns follow evolution rules so that overpowering monsters don't appear until your evolution rate factor says so

### Actual behavior
Special zombies, especially the bionic zombie supersoldiers, start spawning right away--even when the evolution rate is set slower than normal.

### Steps to reproduce the behavior
Start a game, wander around, wait to be ganked by a Zombie Super BAMRU.

### PR DESCRIPTION
**Pull request by Firestorm01X2**
This PR aims to remove very advanced zombie monsters from the beginings of the game and let basic zombies evolve to advanced using evolution rules.

Long time ago evolution was impleneted. But monstergrups still contains advanced monsters in the beginning. 

That may produce advanced mosnters at the start of the game. It is spesially noticabble with wander spawn enabled. 

There is issue based on it:
https://github.com/CleverRaven/Cataclysm-DDA/issues/23063

**For now it is going to be mod**

#### Summary

SUMMARY: Content "MOD: Changing monstergroup to remove very advanced zombie monsters from the beginings ot the game"

#### Purpose of change

Fixes #23063

https://github.com/CleverRaven/Cataclysm-DDA/issues/23063#issuecomment-392184567
>Horde monsters (aka wandering zombies) are spawned via the monster group "GROUP_ZOMBIE", which contains shocker, brute, master, hulk etc. (see "data/json/monstergroups.json").

>And the monsters are upgraded normally when spawned.

>Maybe the monster group should be changed to not contain advanced monsters? Upgrading the monsters upon spawning them is done for probably all spawning monsters, so changing that would have a larger effect.

https://www.reddit.com/r/cataclysmdda/comments/aefhos/did_the_city_layout_rework_change_the_spawn_or/
>kevingranade
>https://github.com/CleverRaven/Cataclysm-DDA/issues/23063 tl;dr we never actually completed the evolution migration where only vanilla zombies spawn at game start, so in particular with wandering zombies enabled you can randomly get more advanced zombies.

#### Describe the solution

**For now it is going to be mod**

Let follow some basic rules. In the beginning there is should be only non heavily mutated zombies.
Zombie count as "non heavily mutated" as long as it is not contain feature that is not common to human host. 

So normal zombes, slow zombies, fast zombies, tought zombes, profession zobmies, zombies wih malfunctoned CBM, zombes with grenades and flametwhowers, zombes with power armor are allowed.

But as long as zombie start develop things, like- bone exoskeleton, acid spewing, supernatural abilites then it is moved from beginning of the game. Evoltution will bring it as time come.

So no acid zombies, hulks and brutes, zombes with extra long arms, exta sharp claws, invisiblility in the darkness etc,. Player will see them later because of evolution process.

But there are exeptions:
* Lab spawns are untouched. It is expected to have advanced zombies there from the start due to "expriements".
* There are special groups to spawn stong enemies. This groups are left untouched.

Important note: Obviously, it will make beginning of the game easier. 

#### Describe alternatives you've considered

Change only GROUP_ZOMBIE (default monster group) and leave others as it was before.

#### Additional context

None.

### CODE REVIEW

**Comment by Firestorm01X2** [comment]
Jeknins, rebuild.

**Comment by Firestorm01X2** [comment]
Fron now ony "GROUP_ZOMBIE" is changed. It is already should recude change of getting endanced zombies ith the beginning with wander spawn. Should I continue?

**Comment by Firestorm01X2** [comment]
After some thinking I made few exception of non mutated rules.

**Comment by Inglonias** [comment]
While this looks to be a nice addition to the game, how I feel about it depends on how monster evolution actually works under the hood. If, as you say, this change doesn't affect things later on, then I'm for it. If the game requires that zombies stay in your reality bubble to evolve, then I'm against it in the mainline. Because of it's JSON nature, this would make a good mod, though.

**Comment by Firestorm01X2** [comment]
>actually works under the hood. If, as you say, this change doesn't affect things later on, then I'm for it. If the game requires that zombies stay in your reality bubble to evolve, then I'm against it in the mainline. Because of it's JSON nature, this would make a good mod, though.

I've started next summer. This way I've got upgraded monsters right of the way.

**Comment by Firestorm01X2** [comment]
I suppose I have to change default evoltion factor. Waiting 200 days to evolve is very long.

**Comment by Vasyan2006** [comment]
This is not a bugfix but a rebalance, because it will be much easier to pillage cities at the game start. This change will be similar to using "Mundane Zombies" mod during first months. I suggest to make it as optional rebalance mod or make a mod to restore old settings.

**Comment by Vasyan2006** [comment]
> change default evolution factor from 4.00 to 1.00
In this case we will have much more hulks after few months. So game start will be easier, but difficulty will increase 4.00 times faster.

**Comment by kevingranade** [comment]
The plan is for population numbers to be higher but the game starts with more mundane zombies at the start so that individuals are easier to deal with.  Then over time evolution kicks in and ratchets up the difficulty.
This is going to be a rather involved balancing process, so I'm not going to push it in before 0.D.

**Comment by Night-Pryanik** [comment]
The game starts at least five days after the Cataclysm. I think this time is more than enough for at least some of the monsters to evolve. Completely removing advanced monsters from the beginning of the game looks like a bit of extreme to me.

**Comment by Firestorm01X2** [comment]
I've decided to create a mod. However it is recommended to play using lower evolution factor (fater evolution). 
Probably is better  to have at least on 2.0 (default is 4.0)

**Comment by Firestorm01X2** [comment]
Jenkins, rebuild

File: data/mods/Alternative_Zombie_Evolution/modinfo.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[
+  {
+    "type": "MOD_INFO",
+    "ident": "alternative_zombie_evolution",
+    "name": "Alternative Zombie Evolution",
+    "authors": [ "Firesorm01X2" ],
```
Comment (Night-Pryanik) [inline data/mods/Alternative_Zombie_Evolution/modinfo.json]:
```suggestion
    "authors": [ "Firestorm01X2" ],
```

File: data/mods/Alternative_Zombie_Evolution/modinfo.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    "maintainers": [ "Firesorm01X2" ],
```
Comment (Night-Pryanik) [inline data/mods/Alternative_Zombie_Evolution/modinfo.json]:
```suggestion
    "maintainers": [ "Firestorm01X2" ],
```

File: data/mods/Alternative_Zombie_Evolution/modinfo.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+[
+  {
+    "type": "MOD_INFO",
+    "ident": "alternative_zombie_evolution",
+    "name": "Alternative Zombie Evolution",
```
Comment (Night-Pryanik) [inline data/mods/Alternative_Zombie_Evolution/modinfo.json]: I think the name is confusing. It doesn't alternates evolution, it just forbids evolved monsters to spawn in the beginning. Consider naming it something like "No advanced monsters in the beginning".

**Comment by Firestorm01X2** [comment]
I think I have add something in code. 
This: https://github.com/CleverRaven/Cataclysm-DDA/pull/14077 have it is own points, but I want true "half life" time.

**Comment by Firestorm01X2** [comment]
Current overview:
* All advanced zombies removed from starting street city groups and house groups.
* Added ability to enable back true half life evolution time, that was changed here: https://github.com/CleverRaven/Cataclysm-DDA/pull/14077 . So now time to evolve will be more random, without unnecessary 100% pass interval.
* Instead added JSON value for guranteed time to evolve to prevent too rapid evolution of one mosnter.
* Base street monster group ("GROUP_ZOMBIE") no longer will be trnasformed to another group "GROUP_ZOMBIE_MID" after some time. Zombies now have to acutally evolve.
* Removed one way shorcut to hul and skeletal juggernouts. To get here zombes should have evolve at least 2 times.
* Time to monster upgrade going to be set around 1 month. Current are way to scattered- between 10 days and month. It is hard to balance due random evlution roll. Also it will give explainable scaling withWORLD OPTION.

New EXTERNAL OPTIONS:

```
{
    "type": "EXTERNAL_OPTION",
    "name": "MONSTER_UPGRADE_TRUE_HALFLIFE",
    "info": "Enables real half life upgrade factor.  If true then minimum ugrade time can be randomly selected as zero.  If false then minimum ugrade time is always at least half_life.",
    "stype": "bool",
    "value": true
  },
  {
    "type": "EXTERNAL_OPTION",
    "name": "MONSTER_UPGRADE_MIN_HALFLIFE",
    "info": "Minimal  half time for monster upgrade.  If monster upgrade halflife time stat is lower than that value then will be used that value.  This value scales with MONSTER_UPGRADE_FACTOR.",
    "stype": "int",
    "value": 50
  },
  {
    "type": "EXTERNAL_OPTION",
    "name": "MONSTER_UPGRADE_MIN_DAYS",
    "info": "Minimal days for monster upgrade.  Monster upgade time in days can't be lower than that no matter what.  This value doesn't scales with MONSTER_UPGRADE_FACTOR",
    "stype": "int",
    "value": 10
}
```

Overall idea:
* Make evolution a lot more unpredictable but at the same time more smooth.
* Advanced monsters no longer going to just appear, but have to evolve from basic forms.
* First mounth you have to deal with only mundane zombies mostly. Hulk actually can happen in firts month but it should be very unlikely.

**Comment by Firestorm01X2** [comment]
Average upgrade time by default (evolution factor 4) set to 2 months.

**Comment by Firestorm01X2** [comment]
I have to end somewhere.
Ready.

**Comment by Firestorm01X2** [comment]
Then it come to mainline evolution times for each monster should be changed.
For now I am planning 45+ old value. It should give average evolution time of 2 months.

**Comment by Firestorm01X2** [comment]
Redoing as minline here: https://github.com/CleverRaven/Cataclysm-DDA/pull/28797

**Comment by Firestorm01X2** [comment]
Likely it is conflicts with this:
https://github.com/CleverRaven/Cataclysm-DDA/pull/28090

So something has to be decided.

**Pull request closed without merging**